cinema/views.py

```diff
@@ -101,8 +105,27 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "create":
+            return MovieCreateSerializer
+
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
The `upload_image` action in `MovieViewSet` is designed to handle image uploads for movies. Ensure that the file upload is properly validated and that any potential errors are handled gracefully. Consider adding checks for file size and type if necessary.